### PR TITLE
Add ansible-test unit test for webhook plugin

### PR DESCRIPTION
**Pull request by hsong-rh**
Passed test command:
```
ansible-test units --docker default
```

### CODE REVIEW

File: benthomasson/eda/tests/unit/event_source/test_webhook.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from plugins.event_source.webhook import main as webhook_main
+
+
+async def start_server(queue, args):
+    await webhook_main(queue, args)
+
+
+async def post_code(server_task, info):
+    async with aiohttp.ClientSession() as session:
+        async with session.post('http://localhost:8000/payload', json=info) as resp:
```
Comment (mkanoor) [inline benthomasson/eda/tests/unit/event_source/test_webhook.py:14]: @hsong-rh Should this take in the args and create the URL based on port number.

File: benthomasson/eda/tests/unit/event_source/test_webhook.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    server_task.cancel()
+
+
+@pytest.mark.asyncio
+async def test_cancel():
+    queue = asyncio.Queue()
+
+    args = {"host": 'localhost', "port": 8000}
+    plugin_task = asyncio.create_task(start_server(queue, args))
+    asyncio.sleep(1)
```
Comment (mkanoor) [inline benthomasson/eda/tests/unit/event_source/test_webhook.py:30]: @hsong-rh Is this sleep needed? Is it just waiting for the other task to startup

Comment (hsong-rh) [reply benthomasson/eda/tests/unit/event_source/test_webhook.py:30]: The purpose of it is to make sure the `plugin_task` run first

File: benthomasson/eda/tests/unit/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+aiohttp
```
Comment (Alex-Izquierdo) [inline benthomasson/eda/tests/unit/requirements.txt:1]: same as I commented here https://github.com/benthomasson/benthomasson-eda/pull/10#discussion_r946708278

File: benthomasson/eda/tests/unit/event_source/test_webhook.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    cancel_task = asyncio.create_task(cancel_code(plugin_task))
+
+    with pytest.raises(asyncio.CancelledError):
+        await asyncio.gather(plugin_task, cancel_task)
```
Comment (Alex-Izquierdo) [inline benthomasson/eda/tests/unit/event_source/test_webhook.py:34]: I think you are missing here the parameter `return_exceptions=True`
https://docs.python.org/3/library/asyncio-task.html#asyncio.gather

File: benthomasson/eda/tests/unit/event_source/test_webhook.py (hunk 2 above)
Comment (Alex-Izquierdo) [reply benthomasson/eda/tests/unit/event_source/test_webhook.py:30]: I have not looked in the deep the code, but in a first look seems that using sleeps to ensure the order execution in a concurrent context seems a bad idea, I'm afraid to have a flaky test.

**Review by Alex-Izquierdo — CHANGES_REQUESTED** [review]

File: benthomasson/eda/tests/unit/requirements.txt (hunk 3 above)
Comment (Alex-Izquierdo) [reply benthomasson/eda/tests/unit/requirements.txt:1]: I'm trying to do it for kafka: https://github.com/benthomasson/benthomasson-eda/pull/12/files

File: benthomasson/eda/tests/unit/event_source/test_webhook.py (hunk 4 above)
Comment (hsong-rh) [reply benthomasson/eda/tests/unit/event_source/test_webhook.py:36]: No, I need it to raise the cancelled error, which is the default `return_exceptions=False` means

File: benthomasson/eda/tests/unit/requirements.txt (hunk 3 above)
Comment (hsong-rh) [reply benthomasson/eda/tests/unit/requirements.txt:1]: Based on the doc: https://docs.ansible.com/ansible/latest/dev_guide/testing_units.html, the requirements.txt has to be located at tests/unit directory

File: benthomasson/eda/tests/unit/event_source/test_webhook.py (hunk 2 above)
Comment (hsong-rh) [reply benthomasson/eda/tests/unit/event_source/test_webhook.py]: It's been removed

File: benthomasson/eda/tests/unit/event_source/test_webhook.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    await webhook_main(queue, args)
+
+
+async def post_code(server_task, info):
+    url = f'http://{info["host"]}/{info["endpoint"]}'
+    payload = info["payload"]
+
+    async with aiohttp.ClientSession() as session:
+        async with session.post(url, json=payload) as resp:
+            print(resp.status)
```
Comment (mkanoor) [inline benthomasson/eda/tests/unit/event_source/test_webhook.py:18]: @hsong-rh Should we raise an exception here if the status is not 200.

Comment (hsong-rh) [reply benthomasson/eda/tests/unit/event_source/test_webhook.py:18]: @mkanoor If we raise an exception here, when it really happens, the next line `server_task.cancel()` will not be called. The server will keep running and never back.

**Review by benthomasson — APPROVED** [review]

**Review by mkanoor — APPROVED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]

**Pull request merged**